### ISSUE TITLE
Revamp level "auto tick rate" handling

### ISSUE THREAD

**Issue by dktapps** [pmmp/PocketMine-MP#2665]
In most cases I have observed that the "auto tick rate" feature of PocketMine-MP simply causes more problems than it solves.

The tick rate shift usually kicks in when a lag spike happens. Usually, lag spikes are not long-lived (usually only a single tick, for example if GC kicked in). However, when a lag spike occurs during a level tick, the server reacts by setting the level's "tick rate" to `tickTimeMS / 50`. On a temporary lag spike, this is pointless and just causes useless additional lag. A lag spike causing a single tick to take several multiples of 50ms (let's say 500ms) will cause the level tick rate to be raised to 10 ticks. On top of this, the tick rate is not immediately decreased once the level ticking goes back to normal, but instead decreases one by one, causing lots of annoying console spam as it goes (10, 9, 8, 7, 6, ...).

The only case where this feature would be useful is in the case of _consistent_ lag spanning many ticks. However, if this happens, performance is getting worsened arbitrarily instead of just lagging behind.

TL;DR: This "feature" is causing more problems and apparent lag than it solves. It additionally creates a lot of extra internal complexity. It should be revamped or removed.
